### PR TITLE
feat: show only relevant delivery options

### PR DESCRIPTION
**Pull request by joerivanveen**

### CODE REVIEW

File: Setup/UpgradeData.php [hunk 1]
```diff
@@ -685,6 +685,26 @@ public function upgrade(ModuleDataSetupInterface $setup, ModuleContextInterface
...
+                '`path` LIKE "myparcelnl_magento_instabox_settings/delivery/same_day_active"'
+            );
+
+            $sameDayData = $connection->fetchAll($selectSameDayDelivery) ?? [];
+
+            foreach ($sameDayData as $value) {
+                $path = 'myparcelnl_magento_instabox_settings/delivery/same_day_delivery_active';
+
+                $bind  = ['path' => $path];
+                $where = 'config_id = ' . $value['config_id'];
```
Comment (wthijmen) [inline Setup/UpgradeData.php:703]: concatenation

File: Model/Sales/Repository/PackageRepository.php [hunk 2]
```diff
@@ -286,17 +288,17 @@ public function getAgeCheck(array $products, string $carrierPath): bool
...
-    public function setDigitalStampSettings()
+    public function setDigitalStampSettings(string $carrierPath = self::XML_PATH_POSTNL_SETTINGS): PackageRepository
     {
-        $settings = $this->getConfigValue(self::XML_PATH_POSTNL_SETTINGS . 'digital_stamp');
-        if ($settings === null) {
-            $this->_logger->critical('Can\'t set settings with path:' . self::XML_PATH_POSTNL_SETTINGS . 'digital stamp');
-        }
+        $settings = $this->getConfigValue($carrierPath . 'digital_stamp');
+        if (null === $settings || array_key_exists('active', $settings)) {
+            $this->_logger->critical('Can\'t set settings with path:' . $carrierPath . 'digital stamp');
```
Comment (wthijmen) [inline Model/Sales/Repository/PackageRepository.php:299]: concatenation

File: Model/Sales/Repository/PackageRepository.php [hunk 3]
```diff
@@ -190,24 +190,26 @@ public function setWeightFromQuoteProducts(array $products)
...
      */
-    public function setMailboxSettings()
+    public function setMailboxSettings(string $carrierPath = self::XML_PATH_POSTNL_SETTINGS): PackageRepository
     {
-        $settings = $this->getConfigValue(self::XML_PATH_POSTNL_SETTINGS . 'mailbox');
-
-        if ($settings === null) {
-            $this->_logger->critical('Can\'t set settings with path:' . self::XML_PATH_POSTNL_SETTINGS . 'mailbox');
-        }
+        $settings = $this->getConfigValue($carrierPath . 'mailbox');
```
Comment (wthijmen) [inline Model/Sales/Repository/PackageRepository.php:200]: concatenation

File: Model/Rate/Result.php [hunk 4]
```diff
@@ -212,25 +262,23 @@ private function hasMyParcelRate(string $settingPath): bool
...
         }
-        // Check if there are multiple additional options like signature and only_recipient and check if they are both active.
-        if (count($settingPathParts) === 3) {
-            return $this->hasSettingAdditionalOptions($settingPathParts, $map, $separator);
+
+        foreach ($settingPathParts as $index => $option) {
+            if (0 === $index) {
+                continue;
+            }
+            $fullPath = $map . 'delivery/' . $option . $separator . 'active';
```
Comment (wthijmen) [inline Model/Rate/Result.php:275]: concatenation

File: Model/Rate/Result.php [hunk 5]
```diff
@@ -212,25 +262,23 @@ private function hasMyParcelRate(string $settingPath): bool
...
-        // Check if delivery, morning or evening delivery are active.
-        if ($activeDelivery && ('morning' === $settingPathParts[0] || 'evening' === $settingPathParts[0])) {
-            return (bool) $this->myParcelHelper->getConfigValue($map . $settingPathParts[0] . '/active');
-        }
-        // Check if the setting has an additional option like signature or only_recipient and see if it is active.
-        if (count($settingPathParts) === 2) {
-            return $this->hasSettingAdditionalOption($settingPathParts, $map, $separator);
+        $baseSetting = $settingPathParts[0];
+
+        if (! $this->myParcelHelper->getConfigValue($map . $baseSetting . '/active')) {
```
Comment (wthijmen) [inline Model/Rate/Result.php:267]: concatenation

**Review by wthijmen — APPROVED** [review]

File: Model/Rate/Result.php [hunk 6]
```diff
@@ -164,25 +170,69 @@ private function addMyParcelRates(Method $parentRate): void
...
+     * @param  string $packageType
+     * @param  string $fromOptionPath
+     *
+     * @return bool
+     */
+    private function isRelevantOption(string $packageType, string $fromOptionPath): bool
+    {
+        switch ($packageType) {
+
+            case 'letter':
```
Comment (EdieLemoine) [inline Model/Rate/Result.php:219]: gebruik consts van de sdk

File: Model/Rate/Result.php [hunk 7]
```diff
@@ -212,25 +262,23 @@ private function hasMyParcelRate(string $settingPath): bool
...
-            return $activeDelivery;
-        }
-        // Check if delivery, morning or evening delivery are active.
-        if ($activeDelivery && ('morning' === $settingPathParts[0] || 'evening' === $settingPathParts[0])) {
-            return (bool) $this->myParcelHelper->getConfigValue($map . $settingPathParts[0] . '/active');
-        }
-        // Check if the setting has an additional option like signature or only_recipient and see if it is active.
-        if (count($settingPathParts) === 2) {
-            return $this->hasSettingAdditionalOption($settingPathParts, $map, $separator);
+        $baseSetting = $settingPathParts[0];
```
Comment (EdieLemoine) [inline Model/Rate/Result.php:265]: hmmmmmmmmmmm

**Review by EdieLemoine — DISMISSED** [review]

**Review by EdieLemoine — APPROVED** [review]

**Review by wthijmen — APPROVED** [review]

**Pull request merged**